src/main/java/com/instaclick/pentaho/plugin/amqp/AMQPPlugin.java

```diff
@@ -100,6 +162,14 @@ public boolean processRow(StepMetaInterface smi, StepDataInterface sdi) throws K
 
                 setOutputDone();
 
+                //now output rowset closed, so we can wait for our confirmation steps
+                if ( data.activeConfirmation )  {
+                    while (!isStopped() && data.watchedConfirmStep.size() > 0) {    
+                        Thread.sleep(500);

```

Comment:
i've tried replicate blockuntilstep finished logic here, with some changes, original kettle step check  steps itself, but our plugin relay on event received for StepListener interface StepFinished event